### ISSUE TITLE
Using tfrecord + dataset API  + estimator, how to add dynamic input to randomly augment data like in old version

### ISSUE THREAD

**Issue by LZDSJTU** [tensorflow/tensorflow#21383]
I am now using tensorflow 1.9 and I want to use tfrecord + dataset API  + estimator.
However, I find a question.
For example, I use tfrecord to record the input image and label. However, in each training step, I want to generate random indices to delete the value of some pixels. The indices is generated in each step.

Before I used old version tensorflow, I can do the following
placeholder 1 : image from dataset
placeholder 2: label from dataset
placeholder 3: indices tf.random
And I can use tf.gather_nd(image, indice=inidices), so I can randomly augment the input image.

However, I do not know how to feed the random numpy array in tensorflow 1.9

**Comment by LZDSJTU** [comment]
I use tf_record to record my data and label. However, I also want to use a random input which cannot be recorded by tf_record. Usually, in old tensorflow version, we can use the following code:

image = tf.placeholder(tf.float32, shape=(None, height,width,3))
indices = tf.placeholder(tf.float32, shape=(None, height,width))
image_filter = tf.gather_nd(image, indices)
And here, the indices will be fed a value of

indice = np.random.rand()
which is changeable and cannot be recorded before training. If using `

sess.run(......., feed_dict = {.......,'indices':indice})
this can be very easy to feed the indices.

However, I do not know how to implement this while using estimator.train as I do not know where to add the definition of indices and where to feed it into the network.

**Comment by mpekalski** [comment]
That is not a place for usage questions.
Please do it for example on stackoverflow.com

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tatianashp** [comment]
@LZDSJTU So sorry this fell through the cracks. If you still have this question please ask it on StackOverlow. [Stack Overflow](http://stackoverflow.com/questions/tagged/tensorflow)
